### ISSUE TITLE
Anonymous reporting for Blacklisted domains

### ISSUE THREAD

**Issue by Anushi1998** [webcompat/webcompat.com#2688]
@karlcow  The message in [config/environment.py](https://github.com/webcompat/webcompat.com/blob/master/config/environment.py#L82) should not use the word `Anonymous` as I still can't create the issue when I opt to share my Github Id.

**Comment by karlcow** [comment]
This is probably a bug on our side. 
https://github.com/webcompat/webcompat.com/blob/e55d2ff4f5f3fe34ed3187bcc3e855a8d0338551/config/environment.py#L82-L85

We should allow the reporting for identified users. 
https://github.com/webcompat/webcompat.com/blob/ede2c5dca9c5755baaa7c07e29d59d5f4fc461cb/webcompat/views.py#L233-L237

so basically the condition should probably be `if is_blacklisted_domain(…) and not g.user:`

I changed the title.
Let's see if @miketaylr agrees with me.

**Comment by miketaylr** [comment]
>  Let's see if @miketaylr agrees with me.

Yep, makes perfect sense. If people abuse things, we can ban them for a day, week, month, or forever.
